warehouse/integrations/snowflake/snowflake.go

```diff
@@ -181,6 +193,8 @@ func New(conf *config.Config, log logger.Logger, stat stats.Stats) *Snowflake {
 	sf.config.debugDuplicateTables = lo.Map(conf.GetStringSlice("Warehouse.snowflake.debugDuplicateTables", nil), func(item string, index int) string {
 		return strings.ToUpper(item)
 	})
+	sf.config.privileges.fetchSchema.required = conf.GetStringSlice("Warehouse.snowflake.privileges.fetchSchema.required", []string{"USAGE"})
```

Comment:
are we looking to override this config? Just want to understand why this is a slice. I am assuming for the current snowflake implementation, `USAGE` is what we need

Reply:
Yeah for now USAGE is sufficient. Made it configurable, just in case if it is needed.

Reply:
nit: we will be forced to change the default value and check for extra grants if our implementation changes. I think we might never use this config. Is this done with any other warehouse previously?

Reply:
No. This is the first time we are introducing grants-based privileges for verifying the validations. There is no harm in making this configurable.